token/program-2022/src/pod.rs

```diff
@@ -117,6 +185,31 @@ impl From<PodBool> for bool {
     }
 }
 
+#[cfg(feature = "serde-traits")]
+impl Serialize for PodBool {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+        where
+            S: Serializer,
+    {
+        let serde_bool = bool::from(self);
+        serde_bool.serialize(serializer)
+    }
+}
+
+#[cfg(feature = "serde-traits")]
+impl<'de> Deserialize<'de> for PodBool {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+        where
+            D: Deserializer<'de>,
+    {
+        let serde_bool = bool::deserialize(deserializer)?;
+        Ok(PodBool::from(serde_bool))
+    }
+}
```

Comment:
Rather than implementing serialize and deserialize by hand, would it be possible to use a `from` and `into` bound whenever the field is encountered? ie:

```
#[serde(from = "bool", into = "bool")] 
my_field: PodBool
```

Reply:
Good catch, updated code, please check